### PR TITLE
Update release-checklist.md

### PR DESCRIPTION
**Pull request by armenzg**
Some updating of the notes and some polishing.

You can see the rendered version in [here](https://github.com/getsentry/action-release/blob/ef7c320f10c6cfb2e386ed3dd279eae127a8f3ab/.github/ISSUE_TEMPLATE/release-checklist.md).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 1]
```diff
@@ -1,27 +1,47 @@
...
+     - [ ] Update the Full Changelog link to use the right tags (there's a minor bug)
+        - The first value is the last release; the second value is the tag that has not yet been created, thus, the link won't be working yet
 - [ ] Once published, visit [the marketplace](https://github.com/marketplace/actions/sentry-release) to see that the release is showing up
-- Update existing major tag (i.e. v1, v2 etc)
-  - [ ] Run `git tag -f v<Major Release Num>`
-  - [ ] Run `git push origin master --tags -f`
+- [ ] Update existing major tag (i.e. v1, v2 etc); Run the following commands
+  - [ ] `git checkout master && git pull`
+  - [ ] `git tag -f v<Major Release Num>`
+  - [ ] `git push origin master --tags -f`
```
Comment (scttcper) [inline .github/ISSUE_TEMPLATE/release-checklist.md:28]: you can make a new tag at the same time as drafting a release in the github ui

**Review by scttcper — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 1 above)
Comment (scttcper) [reply .github/ISSUE_TEMPLATE/release-checklist.md:28]: mostly force pushing to master is concerning

Comment (armenzg) [reply .github/ISSUE_TEMPLATE/release-checklist.md:28]: This is different. We create a tag for a release *but* we also have to move forward the v1 tag (which cannot be done via the UI AFAIK)

Comment (armenzg) [reply .github/ISSUE_TEMPLATE/release-checklist.md:28]: We're only force pushing tags and nothing else

**Pull request merged**